.github/workflows/main_push_and_pull_request_workflow.yml

```diff
@@ -24,13 +24,62 @@ jobs:
           java-version: ${{ matrix.java-version }}
           distribution: temurin
 
+      - name: Read version from properties file
+        id: read-version
+        run: |
+          version=$(grep -oP 'version=\K[^[:space:]]+' gradle.properties)
+          echo "Version found: $version"
+          echo "VERSION=$version" >> $GITHUB_ENV
+
       - name: Build with Gradle
-        run: ./gradlew build distTar -x integrationTest
+        run: make build/distributions/tiered-storage-for-apache-kafka-${{ env.VERSION }}.tgz
 
       - name: Run integration tests
-        run: ./gradlew integrationTest
+        run: make integration_test
+
+      - name: Upload Build Artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: build-${{ matrix.java-version }}
+          path: build/distributions/*.tgz
+          if-no-files-found: error
```

Comment:
I guess it's just a check that the tgz has been built, right? If'm thinking if we can be OK without it, only adding `if-no-files-found: error` to the e2e section below...

Reply:
As the output is required on the following job, I was thinking failing fast may be preferred -- though we can drop this and fail later.

Reply:
Decided to remove this as validation is handled at the Makefile level (worst case it rebuilds the binaries if not found).